### ISSUE TITLE
GDScript 'match' statment fails when exporting in release

### ISSUE THREAD

**Issue by lyubomirv** [godotengine/godot#71577]
### Godot version

4.0 beta12

### System information

Linux x86-64

### Issue description

Having the following **valid** gdscript code in a file causes an issue when the project is exported in release:

```gdscript
func some_func(x: int) -> bool:
    match x:
        0:
            return false
        _:
            return true
```

This is completely valid according to the documentation - we have one case for `0` and then a catch-all case. Now comes the weird part. If the project is exported in debug (pick 'Export with Debug' during the export) then everything is fine and the project works as expected.

However, if the project is exported in release (_not_ picking 'Export with Debug), then there is a weird behaviour when running the game. It took me a while to find out why but when I finally ran the exported game from the command line on Linux, I noticed this in the log:

> SCRIPT ERROR: Parse Error: Not all code paths return a value.
>           at: GDScript::reload (res://test_scene.gd:10)

The log message points to the `match` statement above. Apparently the gdscript file containing this failed to compile properly because of this error.

This log message does **NOT** appear when the game is exported in debug.

Adding another `return` statement at the end of the function fixes the problem. Still, this is supposed to be valid code, as the default `_` case should be considered enough for the function.

### Steps to reproduce

Tested with exporting for linux x86-64 only!

1. Get the minimal project
It includes a simple scene with godot's icon in the centre, which is set to be invisible in the scene.
The scene has an attached script that decides in the `_ready()` function whether to make the icon visible by using a `match` statement as explained above.

2. Export the project for linux x86-64 in release

3. See that the exported game doesn't show the icon in the centre of the game window
You can also run the game from the command line to see the log message

4. Export the game again, this time in debug

5. See that the exported game now shows the icon in the centre of the window

### Minimal reproduction project

[test_gdscript_match.tar.gz](https://github.com/godotengine/godot/files/10436439/test_gdscript_match.tar.gz)

**Comment by haydenwv** [comment]
I downloaded the test project, opened it with 4.0 beta 12, and exported the project for Windows Desktop x86_64 in both debug and release modes. I ran the debug build, and the icon appeared as expected. I ran the release build (from command line), and the icon didn't appear, and the log showed

    SCRIPT ERROR: Parse Error: Not all code paths return a value.
              at: GDScript::reload (res://test_scene.gd:16)
              
I also did the same with 4.0 beta 11, and with 4.0 beta 13, and had identical results.
